app/src/main/java/org/fossasia/openevent/app/core/auth/login/LoginFragment.java

```diff
@@ -50,24 +56,43 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void onStart() {
         super.onStart();
         getPresenter().attach(this);
-        binding.setLogin(getPresenter().getLogin());
-        getPresenter().start();
+
+        //attaching the view model
+        loginFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(LoginViewModel.class);
+        handleEmails();
+
+        //getPresenter().start();
+        binding.setLogin(loginFragmentViewModel.getLogin());
+
         binding.btnLogin.setOnClickListener(view -> {
             if (!validator.validate())
                 return;
 
             ViewUtils.hideKeyboard(view);
 
             String url = binding.url.baseUrl.getText().toString().trim();
-            getPresenter().setBaseUrl(url, binding.url.overrideUrl.isChecked());
-            getPresenter().login();
+            loginFragmentViewModel.setBaseUrl(url, binding.url.overrideUrl.isChecked());
+            loginFragmentViewModel.login().observe(this, this::onResponse);
         });
 
         binding.signUpLink.setOnClickListener(view -> openSignUpPage());
-
         binding.forgotPasswordLink.setOnClickListener(view -> openForgotPasswordPage());
     }
 
+    private void handleEmails() {
+        emailsList = loginFragmentViewModel.getEmailList();
+        if(emailsList != null)
+            binding.emailDropdown.setAdapter(
+                new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, new ArrayList<>(emailsList)));
+    }
+
+    private void onResponse(Boolean response) {
+        String message = response ? "Successful Login!" : "UnSuccessful Login!";
+        Toast.makeText(getActivity(), message ,Toast.LENGTH_SHORT).show();
+        startActivity(new Intent(getActivity(), MainActivity.class));
+        getActivity().finish();
+    }
```

Comment:
Message as String should be sent by the LiveData, view shouldn't have any logic. Similarly, opening the activity should be sent by ViewModel as well

Reply:
Viiew will handle the logic definitely as it was handling it before. But rather than having a livedata encapsulate a pair, it should be two separate live data

Reply:
Why shouldn't we use Pair? I mean is there any downside to it.

Reply:
It is independent from each other. There is no logical coupling between them